### PR TITLE
Leverage miner to do block building for the builder API

### PR DESCRIPTION
**Pull request by ferranbt**
This PR proposes using the `miner` module from `geth` as the core primitive for the builder api session. Before, the builder was completely separated and had its own logic to handle transaction processing, reverts, keeping track of generated receipts, building the block. Like we have seen with the Deneb fork, we should as much as possible leverage geth logic when building blocks. This PR addresses that.

### CODE REVIEW

File: miner/builder_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"testing"
+
+	"github.com/ethereum/go-ethereum/consensus/clique"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBuilder_AddTxn_Simple(t *testing.T) {
```
Comment (lthibault) [inline miner/builder_test.go:14]: Can we add `t.Parallel()` to the top of this test (idem for the others)?  Our test suite is quite lengthy as it is and I'm keen to cut down on its execution time where possible.

**Review by lthibault — COMMENTED** [review]
I am 100% on board with the overall direction of the PR.  Very good idea!
I've flagged a small opportunity for improving the tests inline.

I'll review again when CI passes, but LGTM so far 👍

**Comment by dmarzzz** [comment]
Is the idea of this PR that we will be sending the blocks from the execution node to the mev-boost relay?

**Comment by ferranbt** [comment]
@lthibault The PR is ready for review.

**Review by lthibault — APPROVED** [review]

**Pull request merged**